assets/helpers/paymentIntegrations/newPaymentFlow/visitToken/reducer.js

```diff
@@ -0,0 +1,41 @@
+// @flow
+
+// ----- Imports ----- //
+
+import type { Action } from './actions';
+
+
+// ----- Types ----- //
+
+export type VisitToken = {
+    token: ?string,
```

Comment:
I've had to make it optional, this isn't really optional.  I'm not sure how to tell they type system that `window.guardian.visitToken` will always be defined

Reply:
I think we would need to define a type for `window.guardian`, which we currently don't. The accuracy of this type would be dependent on the serverside logic which plays out that JSON in a `<script>` tag. Maybe something really clever for this would be to generate Flow types from Scala for those links between Twirl & JavaScript... https://github.com/davegurnell/bridges??

Reply:
nice idea. that would be a good thing to do in future.  The writing into the page is effectively a serialisation to string, so all the information is lost.  Having to only respecify this at the boundary wouldn't be a disaster, but if we could not lose it in the first place it would be better.